src/main/scala/piconot/internal/DSL_API.scala

```diff
@@ -0,0 +1,221 @@
+def setWall(dir: MoveDirection): Surroundings = dir match {
+  case North => Surroundings(Blocked, Anything, Anything, Anything)
+  case South => Surroundings(Anything, Anything, Anything, Blocked)
+  case West => Surroundings(Anything, Anything, Blocked, Anything)
+  case East => Surroundings(Anything, Blocked, Anything, Anything)
+  case _ => throw new IllegalArgumentException(s"Invalid input: $dir")
+}
+
+// sets wall to a particular direction and all else to what already existed in a surrounding
+def setWall(sur: Surroundings, dir: MoveDirection): Surroundings = {
+  val (n, e, w, s) = sur match {
+    case Surroundings(n, e, w, s) => (n, e, w, s)
+    case null => (Anything, Anything, Anything, Anything)
+  }
+  dir match {
+    case North => Surroundings(Blocked, e, w, s)
+    case South => Surroundings(n, e, w, Blocked)
+    case West => Surroundings(n, e, Blocked, s)
+    case East => Surroundings(n, Blocked, w, s)
+    case _ => throw new IllegalArgumentException(s"Invalid input: $dir")
+  }
+}
+// sets open to a particular direction and all else to Anything
+def setOpen(dir: MoveDirection): Surroundings = dir match {
+  case North => Surroundings(Open, Anything, Anything, Anything)
+  case South => Surroundings(Anything, Anything, Anything, Open)
+  case West => Surroundings(Anything, Anything, Open, Anything)
+  case East => Surroundings(Anything, Open, Anything, Anything)
+  case _ => throw new IllegalArgumentException(s"Invalid input: $dir")
+}
+
+// sets open to a particular direction and all else to what already existed in a surrounding
+def setOpen(sur: Surroundings, dir: MoveDirection): Surroundings = {
+  val (n, e, w, s) = sur match {
+    case Surroundings(n, e, w, s) => (n, e, w, s)
+    case null => (Anything, Anything, Anything, Anything)
+  }
+  dir match {
+    case North => Surroundings(Open, e, w, s)
+    case South => Surroundings(n, e, w, Open)
+    case West => Surroundings(n, e, Open, s)
+    case East => Surroundings(n, Open, w, s)
+    case _ => throw new IllegalArgumentException(s"Invalid input: $dir")
+  }
+}
+// defines how/what happens when a particular keyword is used in the string
+def getWords(str: String): Unit = str match {
+  case "start" => {
+    if nextState == null then state = (setState(0), state._2)
+    if (state._2 == null && state._1 != null) then state = (state._1, state._1)
+    buildRule(setState(0), surroundings, direction, state._2)
+    update()
+  }
+  case "next" => {
+    state = (prevState, state._2)
+    if (state._2 == null && state._1 != null) then state = (state._1, state._1)
+    buildRule(state._1, surroundings, direction, state._2)
+    update()
+  }
+  case str if nextWords.contains(str) => {
+    setWState(str)
+    if (state._2 == null && state._1 != null) then state = (state._1, state._1)
+    buildRule(state._1, surroundings, direction, state._2)
+    update()
+  }
+  case "until" => untilFlag = true
+
+  case str if movingWords.contains(str) => {
```

Comment:
I feel like this code might have been cleaner if you tried to implement it as an external DSL. In fact, I feel like what you're trying to do is in fact more external than internal as you created new syntax that is not native to Scala, and are parsing it. Therefore, I might recommend that you use the built in parser tool in order to break down strings, rather than doing it imperatively!